src/NzbDrone.Core/Indexers/Definitions/Cardigann/CardigannRequestGenerator.cs

```diff
@@ -722,49 +722,60 @@ public async Task<HttpRequest> DownloadRequest(Uri link)
+                            var response = await HttpClient.ExecuteAsync(request);
 
-                    var results = response.Content;
-                    var searchResultParser = new HtmlParser();
-                    var searchResultDocument = searchResultParser.ParseDocument(results);
-                    var downloadElement = searchResultDocument.QuerySelector(selector);
-                    if (downloadElement != null)
-                    {
-                        _logger.Debug(string.Format("CardigannIndexer ({0}): Download selector {1} matched:{2}", _definition.Id, selector, downloadElement.ToHtmlPretty()));
+                            var results = response.Content;
+                            var searchResultParser = new HtmlParser();
+                            var searchResultDocument = searchResultParser.ParseDocument(results);
+                            var downloadElement = searchResultDocument.QuerySelector(queryselector);
+                            if (downloadElement != null)
+                            {
+                                _logger.Debug(string.Format("CardigannIndexer ({0}): Download selector {1} matched:{2}", _definition.Id, queryselector, downloadElement.ToHtmlPretty()));
 
-                        var href = "";
-                        if (download.Attribute != null)
-                        {
-                            href = downloadElement.GetAttribute(download.Attribute);
-                            if (href == null)
+                                var href = "";
+                                if (selector.Attribute != null)
+                                {
+                                    href = downloadElement.GetAttribute(selector.Attribute);
+                                    if (href == null)
+                                    {
+                                        throw new Exception(string.Format("Attribute \"{0}\" is not set for element {1}", selector.Attribute, downloadElement.ToHtmlPretty()));
+                                    }
+                                }
+                                else
+                                {
+                                    href = downloadElement.TextContent;
+                                }
+
+                                href = ApplyFilters(href, download.Filters, variables);
+                                link = ResolvePath(href, link);
+                            }
+                            else
                             {
-                                throw new Exception(string.Format("Attribute \"{0}\" is not set for element {1}", download.Attribute, downloadElement.ToHtmlPretty()));
+                                _logger.Error(string.Format("CardigannIndexer ({0}): Download selector {1} didn't match:\n{2}", _definition.Id, queryselector, results));
+                                throw new Exception(string.Format("Download selector {0} didn't match", queryselector));
```

Comment:
We probably want to continue in the loop here instead of throwing, this allows the loop to try the next selector if the first didn't work.